### DISCUSSION TITLE
MapRequestPayload causes a weird format of my response when validation error happen

### DISCUSSION THREAD

**Discussion by mikabnm** [symfony/symfony#54620] [Q&A]
Hi guys,

I'm on v.6.3.12 (php 8.1.6)

I'm trying to use MapRequestPayload on my route to validate Credential change. The thing is for some reason when there is a validation error (that I need to handle on my UI) the "violation" array doesnt come out well I have a 0 as a key instead, I dont understand why, I suspected a serialization problem

So the output is like this

```
{
    "type": "https://tools.ietf.org/html/rfc2616#section-10",
    "title": "An error occurred",
    "status": 422,
    "detail": "app.user.old_password.invalid",
    "0":
    {
        "propertyPath": "old_password",
        "message": "app.user.old_password.invalid",
        "code": "2d2a8bb4-ddc8-45e4-9b0f-8670d3a3e290"
    },
    "class": "Symfony\\Component\\HttpKernel\\Exception\\HttpException"
}
```
So I expect a Violation key with the object with property path etc. but I get this "0"

Here is my controller code:
```
#[Route(
    'account/credentials',
    name: 'put_account_credentials',
    methods: ['PUT'],
)]
class PutCredentialsAction extends AbstractController
{
    public function __construct(
        private readonly Security $security,
        private readonly UserAssembler $userAssembler,
        private readonly UserManager $userManager,
        private readonly JsonSerializer $serializer,
        private readonly JWTTokenManagerInterface $JWTManager
    ) {
    }

    public function __invoke(
        Request $request,
        #[MapRequestPayload] UserUpdateCredentialsDto $updatedUserDto,
    ): JsonResponse {
        $currentUser = $this->security->getUser();
        if (null === $currentUser) {
            throw new HttpException(Response::HTTP_FORBIDDEN, 'Not logged');
        }

        /** @var UserDto $userDto */
        $userDto = $this->serializer->denormalize($updatedUserDto, UserDto::class, $this->userAssembler->transform($currentUser));

        $currentUser = $this->userAssembler->reverseTransform($userDto, $currentUser);

        $this->userManager->save($currentUser);

        /** @var UserDto $userDto */
        $userDto        = $this->userAssembler->transform($currentUser);
        $userDto->token = $this->JWTManager->create($currentUser);

        return $this->json($userDto, Response::HTTP_OK, ['user']);
    }
}
```

And DTO:

```
class UserUpdateCredentialsDto
{
    #[Assert\NotBlank]
    #[Assert\Length(min: 6)]
    #[AppAssert\EmailDns]
    #[Groups(['update_credentials_all'])]
    public ?string $email = null;

    #[Assert\NotBlank]
    #[Assert\Length(min: 6)]
    #[Groups(['update_credentials_all'])]
    public ?string $password = null;

    #[SecurityAssert\UserPassword(
        message: 'app.user.old_password.invalid',
    )]
    #[Groups(['update_credentials_all'])]
    public ?string $oldPassword = null;
}
```

I really dont get why it's not formated properly, I m kinda out of leads... Any help would be welcome

**Comment by yceruto** [comment]
Can you please create a small app reproducing this issue? thanks!

**Comment by yceruto** [comment]
It looks like a bug, but I failed to see where that "0" comes from

**Comment by mikabnm** [comment]
Hey @yceruto  thank you for taking the time.
I created a new fresh app from scratch [here](https://github.com/mikamoush-nv/request-validation-app) and I don't reproduce the probleme... 

This is my output: 
```
{
	"type": "https:\/\/symfony.com\/errors\/validation",
	"title": "Validation Failed",
	"status": 422,
	"detail": "title: This value should be of type string.",
	"violations": [
		{
			"propertyPath": "title",
			"title": "This value should be of type string.",
			"template": "This value should be of type {{ type }}.",
			"parameters": {
				"{{ type }}": "string"
			}
		}
	],
```

And that is exactly what i want to get on my current app... I dont know where that 0 come from as well

**Comment by mikabnm** [comment]
Hi @yceruto ,

I found the problem. My app works with api platform. The thing is not all my routes are not made with it, we also have classic Symfony Controllers (for most routes actually), apiplatform covers some entities and we re currently trying to migrate everything (long process since big project already in production).
Well ApiPlatforms has his own ConstraintViolationListNormalizer: it's the culprit. I tried to figure out how to configure apiplatform properly to handle validation error serialization properly but i didnt succeed...
My idea was to decorate that Service in my app's code to support normalization only if request uri contains '/api' (prefix for all apiplatform route). Dont know if its the right thing to do but i got to have my response to be serialiazed properly but I think I broke other things.

Any idea of how to make it better maybe ?
